docs/conceptual/runtime/contracts/ink.md

```diff
@@ -0,0 +1,106 @@
+---
+title: ink!
+---
+
+ink! is a [Rust](https://www.rust-lang.org/)-based embedded domain specific language
+([eDSL](https://wiki.haskell.org/Embedded_domain_specific_language)) for writing
+[Wasm](https://webassembly.org/) smart contracts specifically for the [SRML Contracts module](conceptual/runtime/contracts/contracts_module.md).
+
+## Design
+
+The main goals of ink! are user friendliness, conciseness, and efficiency.
+
+ink! is designed to be as close to the Rust programming language as possible. The language uses
+[attribute macros](https://doc.rust-lang.org/reference/procedural-macros.html#attribute-macros) to
+tag standard Rust structures into understandable contract components.
+
+```rust
+#[ink(...)]
+```
+
+Because ink! follows Rust standards, tools like `rustfmt` and `rust-analyzer` already work out of
+the box.
+
+### Contract Components
+
+ink! should still feel familiar to developers who have programmed using other modern smart contract
+languages. The skeleton of a contract has all of the same components that you might expect:
+
+  * Events
+  * Event Topics
+  * Storage
+  * Constructor Functions
+  * Message Functions
+
+### Custom Types
+
+ink! supports custom type definitions needed to interact with the Substrate runtime. By default,
+ink! has access to all the [primitive Substrate types](conceptual/runtime/primitives.md) like
+`AccountId`, `Balance`, `Hash`, etc...
+
+You can specify and interact with additional types which may be specific to your Substrate runtime
+environment.
+
+### Overflow Safety
+
+Being written in Rust, ink! can provide compile-time overflow/underflow safety. Using a Rust
+compiler configuration, you can specify whether you want to support overflowing math, or if you want
+contract execution to panic when overflows occur. No need to continually import "Safe Math"
+libraries, although Rust also provides [integrated checked, wrapped, and saturated math
+functions](https://doc.rust-lang.org/std/primitive.u32.html).
+
+### Test Environment
+
+ink! provides a built in test environment that can be used to perform off-chain unit testing with
```

Comment:
isn't all unit testing off-chain?

Reply:
Ur probably right about that. So you are saying I should remove off-chain since it is redundant?

Reply:
There is a future plan of do on-chain testing but for now it is implicit that it is off-chain. But it is good to keep it just for making the point.